docs/snippets/npm-module-linking.md

```diff
@@ -0,0 +1,214 @@
+---
+title: "Local Development of ApostropheCMS Extensions"
+detailHeading: "Tutorial"
+url: "/tutorials/local-extension-development.html"
+content: "Learn how to develop and test ApostropheCMS extensions locally without running into ES Module import resolution problems. This guide covers reliable approaches that work with Vite bundling and provides troubleshooting tips for common issues."
+tags:
+  topic: "development"
+  type: tutorial
+  effort: intermediate
+---
+# Local Development of ApostropheCMS Extensions
+
+## Why This Matters
+
+When building custom extensions for ApostropheCMS, you need a reliable way to develop and test them locally before publishing to npm. However, ES Modules in Node.js handle symlinks differently than CommonJS, and ApostropheCMS's use of Vite for bundling creates additional complexity that breaks many traditional local development approaches.
+
+This guide shows you proven methods for local extension development that work reliably with the modern ApostropheCMS toolchain.
+
+## The Problem 🔍
+
+ES Modules have stricter rules for resolving module paths through symlinks compared to CommonJS. Additionally, Vite's dependency resolution system can fail when:
+- Extensions are linked using traditional `npm link`
+- File-based dependencies create circular resolution paths
+- Symlinks don't properly resolve nested dependencies
+
+These issues manifest as import errors, missing dependencies, or build failures that can be frustrating to debug.
+
+## Recommended Solutions ✅
+
+### 1. Manual Symbolic Links (Most Reliable)
+
+Based on extensive testing with ApostropheCMS projects, manual symbolic links provide the most reliable development experience with Vite:
+
+#### For Published Extensions
+
+When working on an extension that's already published to npm:
+
+<AposCodeBlock>
+
+```bash
+# In your ApostropheCMS project directory
+# First, install the extension normally
+npm install @your-org/your-extension
+
+# Create symbolic link (use absolute path)
+ln -s /absolute/path/to/your/local/extension node_modules/@your-org/your-extension
```

Comment:
Should clarify this is a symlink to a git checkout of the extension's repo.

Reply:
Is this command sufficient to remove the folder that npm install already made and replace it with a symlink?